app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -735,6 +736,17 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 }
                 return true;
 
+            case R.id.action_duplicate:
+                loyaltyCard = DBHelper.getLoyaltyCard(database, loyaltyCardId);
+                Intent intent = new Intent(getApplicationContext(), LoyaltyCardEditActivity.class);
+                Bundle bundle = new Bundle();
+                bundle.putInt("id", loyaltyCardId);
+                bundle.putBoolean("duplicateId", true);
+                bundle.putBoolean("update", true);
```

Comment:
Because you set update to true here, the LoyaltyCardEditActivity header text changes from "Add Card" to "Edit Card". However, you're creating a new duplicate, so "Edit Card" doesn't make sense.

The cleanest fix would be to remove update and change all the relevant `if (updateLoyaltyCard)` stuff to `if (updateLoyaltyCard || duplicateFromLoyaltyCardId)` in LoyaltyCardEditActivity.